### ISSUE TITLE
Inconsistent configuration for Redis/SQL address

### ISSUE THREAD

**Issue by julianbrost** [Icinga/icingadb#475]
Redis config uses a single config attribute `address` where you specify `address:port`, whereas for the SQL database, there are separate attribtues `host` and `port`. We might want to unify this before the 1.0 release.

**Comment by julianbrost** [comment]
If we unify this, I suggest going with `host` and `port` for the following reasons:
1. go-redis requires a port in its address configuration, so if we would want to specify a default there, we would have to split this string, see if it contains a port, and if not, add one (and not mess this up with IPv6 literals, though Go has suitable functions for that in the standard library).
2. #479: For PostgreSQL, there are situations where you have to specify a Unix domain socket **and** a port number, supporting this with just a single option would require some custom syntax for it like `/run/postgresql:1234`.

**Comment by Al2Klimov** [comment]
Regarding the status quo I'd (honestly) say

![](https://static01.nyt.com/images/2016/08/05/us/05onfire1_xp/05onfire1_xp-superJumbo-v2.jpg)

but I see no problem in making it tidily. :)

In the latter case I'd prefer joining over splitting, i.e host|port separately.

**Comment by julianbrost** [comment]
Yes, essentially the question is whether we ever want to change this. If we want, we should do so before 1.0.
